src/inference/src/dev/make_tensor.cpp

```diff
@@ -205,6 +205,8 @@ class AllocatedTensor : public ViewTensor {
                          OPENVINO_ASSERT(allocator, "Allocator was not initialized");
                          const auto byte_size = element::get_memory_size(element_type, shape_size(shape));
                          auto data = const_cast<Allocator&>(allocator).allocate(byte_size);
+                         OPENVINO_ASSERT(byte_size == 0 || data != nullptr, "Failed to allocate memory");
```

Comment:
```suggestion
                         OPENVINO_ASSERT(byte_size == 0, "Failed to allocate memory");
```

Reply:
The suggestion is incorrect, skip it.
The best option to verify both condition occurs is add unit tests and catch.